cmd/truss/main.go

```diff
@@ -325,33 +335,30 @@ func readPreviousGeneration(serviceDir string) ([]truss.NamedReadWriter, error)
 }
 
 // simpleFileConstructor has function makeSimpleFile of type filepath.WalkFunc
-// This allows for filepath.Walk to be called with makeSimpleFile and build a truss.SimpleFile
-// for all files in a direcotry
+// This allows for filepath.Walk to be called with makeSimpleFile and build a fileSystem
+// for all files in a directory
 type simpleFileConstructor struct {
```

Comment:
It kind of is a constructor. The way I have seen recommended for walking a
filepath is to use `path/filepath.Walk` with a `path/filepath.WalkFunc`.

So `simpleFileConstructor` has a `WalkFunc` as a method, and that method
modifies the state of `simpleFileConstructor` and in the end the field `files`
of `simpleFileConstructor` are the files that were found. So it is sort of a
constructor.

Now that I know more, I can see I could use a function closure to achieve this
effect and the next time this code is touched that can be done.